src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -340,6 +343,62 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  def updateCluster(userInfo: UserInfo,
+                    googleProject: GoogleProject,
+                    clusterName: ClusterName,
+                    clusterRequest: ClusterRequest): Future[Cluster] = {
+    for {
+      cluster <- internalGetActiveClusterDetails(googleProject, clusterName) //throws 404 if nonexistent
+
+      _ <- checkClusterPermission(userInfo, DeleteCluster, cluster) //throws 404 if no auth //TODO: check a more appropriate action here
```

Comment:
calling out this TODO specifically. really seems like an UpdateCluster action could be useful here

Reply:
added modify action here: https://github.com/broadinstitute/sam/pull/245

Reply:
💯